rclcpp/include/rclcpp/duration.hpp

```diff
@@ -38,10 +38,14 @@ class RCLCPP_PUBLIC Duration
    */
   Duration(int32_t seconds, uint32_t nanoseconds);
 
-  // This constructor matches any numeric value - ints or floats.
+  /// Construct duration from the specified nanoseconds.
+  [[deprecated(
+    "Use Duration::from_nanoseconds instead or std::chrono literals. For example:"
+    "rclcpp::Duration::from_nanoseconds(int64_variable);"
+    "rclcpp::Duration(0ns);")]]
```

Comment:
Nit:

```suggestion
    "rclcpp::Duration(std::chrono::nanoseconds(0));")]]
```

Reply:
`0ns` is valid if you bring `namespace std::chrono_literals` into scope.
If I cannot do `using namespace std::chrono_literals` for some reason (e.g. polluting a header file), I would rather use `rclcpp::Duration::from_nanoseconds(0)` than `rclcpp::Duration(std::chrono::nanoseconds(0))`.

If you think this is confusing for cpp users that don't know about `std::chrono_literals` I can take the suggestion.

Reply:
> If I cannot do `using namespace std::chrono_literals` for some reason (e.g. polluting a header file), I would rather use `rclcpp::Duration::from_nanoseconds(0)` than `rclcpp::Duration(std::chrono::nanoseconds(0))`.
> 
> If you think this is confusing for cpp users that don't know about `std::chrono_literals` I can take the suggestion.

It's not that it is confusing, just that the sentence above explicitly talks about `std::chrono` namespace.  Just for the sake of consistency, I would keep using that namespace.  If users know about the `std::chrono_literals`, then of course they can just use that as well.

But its a nit.  I'll leave it to you whether to fix it or not.

Reply:
Ah, I meant to write `std::chrono_literals` in the beginning of the error message.
See https://github.com/ros2/rclcpp/pull/1432/commits/fa216a67719b2c598f9a2e54a516938403bf98f4.